src/controllers/wallet/atomic-swap/tx-proposal.controller.js

```diff
@@ -345,6 +343,30 @@ async function unlockInputs(req, res) {
   }
 }
 
+/**
+ * Fetches the list of proposals being listened to by this wallet on the Atomic Swap Service
+ */
+async function listenedProposalList(req, res) {
+  const proposalMap = await atomicSwapService.getListenedProposals(req.walletId);
+
+  // Transform the map into an array of proposalIds;
+  const list = [];
+  proposalMap.forEach(proposal => {
+    list.push(proposal.id);
+  });
```

Comment:
Can you have something like this: `const list = Array.from(proposalMap.keys());`? If the keys of the map are the proposal ids, this would be enough, right?

Reply:
Good one-liner! Refactored on [1a648b3](https://github.com/HathorNetwork/hathor-wallet-headless/pull/279/commits/1a648b3fe3b8ab89bd4d767f0e84af18e539bba0)